megamek/src/megamek/common/Coords.java

```diff
@@ -79,9 +79,9 @@ public Coords translated(int dir, int distance) {
     public Coords translated(String dir) {
         int intDir = 0;
 
-        try {
+        if (Character.isDigit(dir.charAt(0))) {
             intDir = Integer.parseInt(dir);
-        } catch (NumberFormatException nfe) {
+        } else {
```

Comment:
a case of "exception oriented programming

Reply:
I bet that improves performance a bit.

Reply:
it is, removes the need for dumping the whole stack and memory.